vscode/oso/client/src/index.ts

```diff
@@ -84,53 +83,36 @@ function polarFilesInWorkspaceFolderPattern(folder: WorkspaceFolder) {
   return new RelativePattern(folder, '**/*.polar');
 }
 
-// Trigger a `didOpen` event for every Polar file in `folder` as a somewhat
-// hacky means of transmitting the initial file system state to the server.
-// Alternatives: we could (A) do a file system walk on the server or (B)
-// `workspace.findFiles()` in the client and then send the list of files to the
-// server for it to load. However, by doing it this way, we delegate the
-// responibility for content retrieval to VS Code, which means we can piggyback
-// on their capabilities for, e.g., loading non-`file://` schemes if we wanted
-// to do that at some point in the future.
+// Trigger a [`didOpen`][didOpen] event for every not-already-open Polar file
+// in `folder` as a somewhat hacky means of transmitting the initial file
+// system state to the server. Alternatives: we could (A) do a file system walk
+// on the server or (B) `workspace.findFiles()` in the client and then send the
+// list of files to the server for it to load. However, by doing it this way,
+// we delegate the responibility for content retrieval to VS Code, which means
+// we can piggyback on their capabilities for, e.g., loading non-`file://`
+// schemes if we wanted to do that at some point in the future.
 //
 // Relevant issues:
 // - https://github.com/microsoft/vscode/issues/15723
 // - https://github.com/microsoft/vscode/issues/33046
+//
+// [didOpen]: https://code.visualstudio.com/api/references/vscode-api#workspace.onDidOpenTextDocument
 async function openPolarFilesInWorkspaceFolder(folder: WorkspaceFolder) {
   const pattern = polarFilesInWorkspaceFolderPattern(folder);
   const uris = await workspace.findFiles(pattern);
-  return Promise.all(uris.map(reloadDocument));
+  return Promise.all(uris.map(openDocument));
 }
 
-// Trigger a [`didOpen`][didOpen] event for `uri`.
+// Trigger a [`didOpen`][didOpen] event if `uri` is not already open.
 //
 // The server uses `didOpen` events to maintain its internal view of Polar
 // documents in the current workspace folder.
 //
 // [didOpen]: https://code.visualstudio.com/api/references/vscode-api#workspace.onDidOpenTextDocument
-async function reloadDocument(uri: Uri) {
+async function openDocument(uri: Uri) {
   const uriMatch = (d: TextDocument) => d.uri.toString() === uri.toString();
-  let doc = workspace.textDocuments.find(uriMatch);
-
-  if (doc) {
-    // If `doc` is already open, cycle its `languageId` from `polar` ->
```

Comment:
So was this logic not actually necessary? Or some other change that made it redundant?

Reply:
Not actually necessary. If a document is "open" in VS Code (the scare quotes are b/c this doesn't necessarily mean the document is open in an editor window — in practical terms it means that a `DidOpen` event has been emitted and a `DidClose` event has not yet been emitted), then we get `DidChange` events every time it's changed. If an un-opened document is changed, our `createChangeWatcher` will notice the change and, in turn, call `openDocument`, which will re-open the document in VS Code, transmitting the updated state (post-change) to our language server.